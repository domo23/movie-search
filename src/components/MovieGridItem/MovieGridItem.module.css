@media screen and (min-width: 1600px) {
  .main {
    flex: 0 1 25%;
    height: 50%;
  }
}

@media screen and (min-width: 1200px) and (max-width: 1600px){
  .main {
    flex: 0 1 33%;
    height: 50%;
  }
}

@media screen and (min-width: 800px) and (max-width: 1200px) {
  .main {
    flex: 0 1 50%;
    height: 50%;
  }
}

@media screen and (min-width: 400px) and (max-width: 800px) {
  .main {
    flex: 0 1 100%;
    height: 50%;
  }
}

@media screen and (max-width: 400px) {
  .main {
    flex: 0 1 100%;
    height: 100%;
  }
}

.main {
    min-height: 400px;
    padding: 15px;
    box-sizing: border-box;
}

.card {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 20px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  border-radius: 5px; /* 5px rounded corners */
  box-sizing: border-box;
}

/* On mouse-over, add a deeper shadow */
.card:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}

.openingCrawl {
    text-align: justify;
    font-style: italic;
}