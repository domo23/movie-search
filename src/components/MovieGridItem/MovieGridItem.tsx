import { MovieData } from "../../models/movie-data";
import styles from "./MovieGridItem.module.css";

type Props = {
  movie: MovieData;
};

function MovieGridItem({ movie }: Props) {
  return (
    <div className={styles.main} data-testid="movie-grid-item">
      <div className={styles.card}>
        <h3>{movie.title}</h3>
        <h5>
          {movie.director}, {movie.release_date}
        </h5>
        <div className={styles.openingCrawl}>{movie.opening_crawl}</div>
      </div>
    </div>
  );
}

export default MovieGridItem;
