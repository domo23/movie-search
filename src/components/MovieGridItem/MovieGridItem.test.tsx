import { render } from "@testing-library/react";
import MovieGridItem from "./MovieGridItem";
import '@testing-library/jest-dom/extend-expect';

const mockData = {
    title: 'Title',
    director: 'Director',
    release_date: '23-06-2021',
    opening_crawl: 'Beginning...'
}

it('movie grid item component snapshot', () => {
    const { getByTestId } = render(<MovieGridItem movie={mockData}></MovieGridItem>);
    const module = getByTestId('movie-grid-item');

    expect(module).toMatchSnapshot();
})

it('should contain msg text', () => {
    const { getByTestId } = render(<MovieGridItem movie={mockData}></MovieGridItem>);
    const module = getByTestId('movie-grid-item');

    expect(module).toHaveTextContent(mockData.title);
    expect(module).toHaveTextContent(mockData.director);
    expect(module).toHaveTextContent(mockData.release_date);
})