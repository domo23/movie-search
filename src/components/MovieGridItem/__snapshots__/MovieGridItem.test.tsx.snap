// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie grid item component snapshot 1`] = `
<div
  class="main"
  data-testid="movie-grid-item"
>
  <div
    class="card"
  >
    <h3>
      Title
    </h3>
    <h5>
      Director
      , 
      23-06-2021
    </h5>
    <div
      class="openingCrawl"
    >
      Beginning...
    </div>
  </div>
</div>
`;
